### PR TITLE
FERC asset fixes

### PR DESCRIPTION
**Pull request by jdangerx**
Some fixes to allow PUDL to build the FERC assets.

* deduplicate facts based on filing date
* some refactoring/renaming

Plus, a little changelog for 1.0 so we can release!

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -20,10 +20,10 @@ license = {file = "LICENSE.txt"}
 dependencies = [
     "pydantic>=1.9,<3",
     "coloredlogs>=14.0,<15.1",
-    "arelle-release>=2.3,<2.13",
+    "arelle-release>=2.3,<3",
     "frictionless>=4.4,<5",
     "sqlalchemy>=1.4,<3",
-    "pandas>=1.5,<2",
+    "pandas>=1.5,<2.1",
```
Comment (zaneselvans) [inline pyproject.toml:26]: Pandas 2.1 has already been released and I'd like to work on getting it working in the main PUDL repo, so could we set this to `pandas>=1.5,<2.2" instead?  Until we get it all working in PUDL the more restrictive version over there will be controlling.

Comment (jdangerx) [reply pyproject.toml:26]: Sure, that's easy enough!

**Review by zschira — CHANGES_REQUESTED** [review]
Nice! Only blocking question is whether we should remove all references to `skip_filings`?

File: src/ferc_xbrl_extractor/xbrl.py [hunk 2]
```diff
@@ -83,6 +84,20 @@ def extract(
...
 
+def _dedupe_newer_report_wins(df: pd.DataFrame, primary_key: list[str]) -> pd.DataFrame:
+    if df.empty:
+        return df
+    unique_cols = [col for col in primary_key if col != "filing_name"]
+    old_index = df.index.names
+    return (
+        df.reset_index()
+        .groupby(unique_cols)
+        .apply(lambda df: df.sort_values("report_date").ffill().iloc[-1])
```
[the remark this answers is not in the document]
Comment (jdangerx) [reply src/ferc_xbrl_extractor/xbrl.py:95]: Yeah, that's exactly right!

File: src/ferc_xbrl_extractor/instance.py [hunk 3]
```diff
@@ -371,11 +382,14 @@ def parse(self, fact_prefix: str = "ferc") -> Instance:
         return Instance(context_dict, instant_facts, duration_facts, self.name)
 
 
-def instances_from_zip(instance_path: Path) -> list[InstanceBuilder]:
+def instances_from_zip(
+    instance_path: Path | io.BytesIO, skip_filings: list[str] | None = None
```
[the remark this answers is not in the document]
Comment (jdangerx) [reply src/ferc_xbrl_extractor/instance.py:386]: Yes!

**Review by zschira — APPROVED** [review]
Looks good!

**Pull request merged**